### PR TITLE
Adding usage examples for common tasks

### PR DESCRIPTION
**Pull request by LysandreJik**
Adding a documentation page detailing usage for common tasks (inference, not training

### CODE REVIEW

File: docs/source/usage.rst [hunk 1]
```diff
@@ -0,0 +1,351 @@
+Usage
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This page shows the most frequent use-cases when using the library. The models available allow for many different
+configurations and a great versatility in use-cases. The most simple ones are presented here, showcasing usage
+for tasks such as question answering, sequence classification, named entity recognition and others.
+
+These examples leverage auto-models, but feel free to modify the code to be more specific and adapt it to your specific
+use-case.
```
Comment (julien-c) [inline docs/source/usage.rst]: What are Auto-Models? (A way to instantiate the correct architecture of model automatically)

File: docs/source/usage.rst [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+Language Modeling
+----------------------------------------------------
+
+Language modeling is the task of fitting a model to a corpus, which can be domain specific. All popular transformer
+based models are trained using a variant of language modeling, e.g. BERT with masked language modeling, GPT-2 with
+causal language modeling.
+
+Language modeling can be useful outside of pre-training as well, for example to shift the model distribution to be
+domain-specific: using a language model trained over a very large corpus, and then fine-tuning it to a news dataset
+or on scientific paper e.g. `LysandreJik/arxiv-nlp <https://huggingface.co/lysandre/arxiv-nlp>`__.
```
Comment (julien-c) [inline docs/source/usage.rst]: *papers

File: docs/source/usage.rst [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    TensorFlow 2.0 and PyTorch.
+    """
+
+    questions = [
+        "How many pretrained models are available in Transformers?",
+        "What does Transformers provide?",
+        "Transformers provides interoperability between which frameworks?",
+    ]
+
+    for i in range(len(questions)):
```
Comment (julien-c) [inline docs/source/usage.rst]: Replace with: for question in questions?

File: docs/source/usage.rst [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+
+Extractive Question Answering
+----------------------------------------------------
+
+Extractive Question Answering is the task of extracting an answer from a text given a question. An example of a
+question answering dataset is the SQuAD dataset, which is entirely based on that task. If you would like to fine-tune
+a model on a SQuAD task, you may leverage the `run_squad.py`.
+
+Here is an example using the pipelines do to sentiment analysis: identifying if a sequence is positive or negative.
+It leverages a fine-tuned model on sst2, which is a GLUE task.
```
Comment (julien-c) [inline docs/source/usage.rst]: I think this is out of place here?

**Review by julien-c — APPROVED** [review]
Nice!

**Comment by LysandreJik** [comment]
I've added a way to switch between `PyTorch` and `TensorFlow` implementations. I didn't want to have a wall of code that had the two frameworks, so now there's a toggle to show which framework you would like to see.

It works as follows: a javascript method parses the documentation page shown and looks for `.highlight` classes, which are the code blocks. In there, it looks for `## PYTORCH CODE`, which represents the beginning of a `PyTorch` snippet and `## TENSORFLOW CODE` which represents the beginning of a `TensorFlow` snippet.

Would love an opinion on the Javascript code as well. Would love to convert this to TS down the road.

Here's a gif of the result
peek3

**Comment by joshchagani** [comment]
@LysandreJik reviewing the JS now and had a question. Depending on your thoughts, would it make sense from a UX standpoint for all of the buttons to toggle together? So, if a user selects "Tensorflow" all of the code blocks would switch to "Tensorflow".

**Comment by LysandreJik** [comment]
I guess this would be cool and makes sense from a UX standpoint. Do you think it's necessary or can it wait for the second version?

**Comment by joshchagani** [comment]
It can 100% wait for a second version.

**Pull request merged**